runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -70,55 +77,58 @@ class MM_ObjectAllocationAPI
 	}
 
 	VMINLINE void
-	initializeContiguousIndexableObject(J9VMThread *currentThread, bool initializeSlots, J9Class *arrayClass, uint32_t size, uintptr_t dataSize, j9object_t *objectHeader)
+	initializeContiguousIndexableObject(J9VMThread *currentThread, bool initializeSlots, J9Class *arrayClass, uint32_t size, uintptr_t dataSize, j9object_t objectHeader)
 	{
+		void *dataAddr = (void *)((uintptr_t)objectHeader + _contiguousHeaderSize);
 		bool isCompressedReferences = J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(currentThread);
-
 		if (isCompressedReferences) {
-			uintptr_t headerSize = sizeof(J9IndexableObjectContiguousCompressed);
-			J9IndexableObjectContiguousCompressed *header = (J9IndexableObjectContiguousCompressed*)*objectHeader;
+			J9IndexableObjectContiguousCompressed *header = (J9IndexableObjectContiguousCompressed *)objectHeader;
 			header->clazz = (uint32_t)(uintptr_t)arrayClass;
 			header->size = size;
-			void *dataAddr = (void *)((uintptr_t)header + headerSize);
 #if defined(J9VM_ENV_DATA64)
-			header->dataAddr = dataAddr;
+			if (_isDataAddrPresent) {
+				((J9IndexableObjectWithDataAddressContiguousCompressed *)header)->dataAddr = dataAddr;
+			}
 #endif /* J9VM_ENV_DATA64 */
-			initializeIndexableSlots(initializeSlots, dataSize, dataAddr);
 		} else {
-			uintptr_t headerSize = sizeof(J9IndexableObjectContiguousFull);
-			J9IndexableObjectContiguousFull *header = (J9IndexableObjectContiguousFull*)*objectHeader;
+			J9IndexableObjectContiguousFull *header = (J9IndexableObjectContiguousFull *)objectHeader;
 			header->clazz = (uintptr_t)arrayClass;
 			header->size = size;
-			void *dataAddr = (void *)((uintptr_t)header + headerSize);
 #if defined(J9VM_ENV_DATA64)
-			header->dataAddr = dataAddr;
+			if (_isDataAddrPresent) {
+				((J9IndexableObjectWithDataAddressContiguousFull *)header)->dataAddr = dataAddr;
+			}
 #endif /* J9VM_ENV_DATA64 */
-			initializeIndexableSlots(initializeSlots, dataSize, dataAddr);
 		}
+		initializeIndexableSlots(initializeSlots, dataSize, dataAddr);
 	}
 
 	VMINLINE void
 	initializeDiscontiguousIndexableObject(J9VMThread *currentThread, J9Class *arrayClass, j9object_t *objectHeader)
```

Comment:
should pass by value (rather by address), same as with Contiguous

Reply:
note that the rest of the code in this method uses it inconsistently (some as by-value, some by-address)